src/components/PeoplePicker/PeoplePicker.tsx

```diff
@@ -1,113 +1,215 @@
+            PrincipalSource: 15,
+            PrincipalType: 1,
+          });
+        let newPersonas: IPersonaProps[] = [];
+        results.forEach((person: IPeoplePickerEntity) => {
+          const persona: SPPersona = {
+            text: person.DisplayText,
+            secondaryText: person.EntityData.Title,
+            imageInitials:
+              person.DisplayText.substr(
+                person.DisplayText.indexOf(" ") + 1,
+                1
+              ) + person.DisplayText.substr(0, 1),
+            Email: person.EntityData.Email,
+          };
+          newPersonas.push(persona);
+        });
+
+        /* No Cache Support Yet 
+        // Create list of matching cached suggestions
+        let cachedResults = cachedPeople
+          .getCachedPeople()
+          .filter((p) =>
+            p.text?.toLowerCase().includes(filterText.toLowerCase())
+          );
+
+        // If we have a cached entry, remove the matching entry from newPersonas, so it is only listed once
+        if (cachedResults && newPersonas) {
+          newPersonas = removeDuplicates(newPersonas, cachedResults);
+        }
+
+        // Return a listing of the cached matching entries, followed by the matching user entries
+        filteredPersonas = [...cachedResults, ...newPersonas];
+        */
+
+        //TODO: Remove this and utilize cache
+        filteredPersonas = [...newPersonas];
+      }
+
+      // If people were already selected, then do not list them as possible additions
+      if (currentPersonas && filteredPersonas) {
+        filteredPersonas = removeDuplicates(filteredPersonas, currentPersonas);
+      }
+
+      if (currentPersonas) {
         filteredPersonas = removeDuplicates(filteredPersonas, currentPersonas);
       }
-      filteredPersonas = limitResults ? filteredPersonas.slice(0, limitResults) : filteredPersonas;
+      filteredPersonas = limitResults
+        ? filteredPersonas.slice(0, limitResults)
+        : filteredPersonas;
       return filterPromise(filteredPersonas);
     } else {
       return [];
     }
   };
 
   const filterPersonasByText = (filterText: string): IPersonaProps[] => {
-    return peopleList.filter(item => doesTextStartWith(item.text as string, filterText));
+    return peopleList.filter((item) =>
+      doesTextStartWith(item.text as string, filterText)
+    );
   };
 
-  const filterPromise = (personasToReturn: IPersonaProps[]): IPersonaProps[] | Promise<IPersonaProps[]> => {
-    if (process.env.NODE_ENV === 'development') {
+  const filterPromise = (
+    personasToReturn: IPersonaProps[]
+  ): IPersonaProps[] | Promise<IPersonaProps[]> => {
+    if (process.env.NODE_ENV === "development") {
       return convertResultsToPromise(personasToReturn);
     } else {
       return personasToReturn;
     }
   };
 
   const onItemsChange = (items: IPersonaProps[] | undefined): void => {
-    if(items)
-    {
+    if (items) {
       setCurrentSelectedItems(items);
+      props.updatePeople(items);
```

Comment:
Looks like you're managing state in two places -- within the component and within the parent. Should probably stick with just the parent.

Reply:
I will look into this as I continue work on making the PeoplePicker component flexible so it can be used across applications with various hooks etc.  Especially with trying to add in validation.

Reply:
We can discuss it further if you want now, or can discuss as I continue to flesh things out.  Part of what I was thinking about having a local component PP state, and it's separate parent state, is that I may not want to update my parent state if the local state has an "Error", such as "Employee cannot be own supervisor".  Would need the PP component to show it in it's controlled component, but wouldn't want the state of the parent to hold that "bad" data.

Reply:
PeoplePicker doesn't need to handle errors directly. It's used only for selecting SP Personas.
State would still come from and be controlled by the parent. Conditional checks can be handle by passing a custom function/callback to the PeoplePicker in place of your setState. That function would have the same signature as the setState. That function would then call setState with the passed persona if there's no error, or run some sort of "oops" path if there is an issue.

Reply:
Created issue #14